### ISSUE TITLE
Can't deploy smart contracts to Rinkeby

### ISSUE THREAD

**Issue by Kaleb47** [trufflesuite/truffle#5390]
### Description

I have altered my `truffle-config.js` like so,

```
const HDWalletProvider = require('@truffle/hdwallet-provider');
const os = require('os');
const key = require("./secrets.json").key;
const apiKey = require("./etherscan.json").apiKey;

module.exports = {


  networks: {


    rinkeby: {
      provider: () => new HDWalletProvider(key, 'https://rinkeby.infura.io/v3/12e1cc7fabf84102b34e6532bb10d78f'),
      network_id: 3,
      gas: 55000000,
      confirmations: 2,
      timeoutBlocks: 200,
      skipDryRun: true
    },
  }
}
```

I ran the following command ```truffle console --network rinkeby```

I received this error:
```
Error: The network id specified in the truffle config (3) does not match the one returned by the network (4).  Ensure that both the network and the provider are properly configured.
    at Object.detectAndSetNetworkId (/home/gitpod/.nvm/versions/node/v16.16.0/lib/node_modules/truffle/build/webpack:/packages/environment/environment.js:114:1)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at Object.detect (/home/gitpod/.nvm/versions/node/v16.16.0/lib/node_modules/truffle/build/webpack:/packages/environment/environment.js:24:1)
    at Object.module.exports [as run] (/home/gitpod/.nvm/versions/node/v16.16.0/lib/node_modules/truffle/build/webpack:/packages/core/lib/commands/console/run.js:19:1)
    at runCommand (/home/gitpod/.nvm/versions/node/v16.16.0/lib/node_modules/truffle/build/webpack:/packages/core/lib/command-utils.js:184:1)
Truffle v5.5.24 (core: 5.5.24)
Node v16.16.0
```

**Comment by sukanyaparashar** [comment]
Hi @Kaleb47. The `network_id` specified in your `truffle-config.js` under `rinkeby` network is **3**, which should be **4** for `rinkeby` network. Please let us know if that resolves your issue. Thanks for raising this.

**Comment by BilliDaGOAT** [comment]
> Hi @Kaleb47. The `network_id` specified in your `truffle-config.js` under `rinkeby` network is **3**, which should be **4** for `rinkeby` network. Please let us know if that resolves your issue. Thanks for raising this.

i changed it to 4. but I have another issue.

```
Compiling your contracts...
===========================
> Everything is up to date, there is nothing to compile.
> Something went wrong while attempting to connect to the network at http://127.0.0.1:9545/. Check your network configuration.
ProviderError: 
Could not connect to your Ethereum client with the following parameters:
    - host       > 127.0.0.1
    - port       > 9545
    - network_id > 4
Please check that your Ethereum client:
    - is running
    - is accepting RPC connections (i.e., "--rpc" or "--http" option is used in geth)
    - is accessible over the network
    - is properly configured in your Truffle configuration file (truffle-config.js)

    at /home/gitpod/.nvm/versions/node/v16.16.0/lib/node_modules/truffle/build/webpack:/packages/provider/wrapper.js:76:1
    at /home/gitpod/.nvm/versions/node/v16.16.0/lib/node_modules/truffle/build/webpack:/packages/provider/wrapper.js:114:1
    at XMLHttpRequest.request.onreadystatechange (/home/gitpod/.nvm/versions/node/v16.16.0/lib/node_modules/truffle/build/webpack:/node_modules/web3-providers-http/lib/index.js:98:1)
    at XMLHttpRequestEventTarget.dispatchEvent (/home/gitpod/.nvm/versions/node/v16.16.0/lib/node_modules/truffle/build/webpack:/node_modules/xhr2-cookies/dist/xml-http-request-event-target.js:34:1)
    at XMLHttpRequest.exports.modules.996763.XMLHttpRequest._setReadyState (/home/gitpod/.nvm/versions/node/v16.16.0/lib/node_modules/truffle/build/webpack:/node_modules/xhr2-cookies/dist/xml-http-request.js:208:1)
    at XMLHttpRequest.exports.modules.996763.XMLHttpRequest._onHttpRequestError (/home/gitpod/.nvm/versions/node/v16.16.0/lib/node_modules/truffle/build/webpack:/node_modules/xhr2-cookies/dist/xml-http-request.js:349:1)
    at ClientRequest.<anonymous> (/home/gitpod/.nvm/versions/node/v16.16.0/lib/node_modules/truffle/build/webpack:/node_modules/xhr2-cookies/dist/xml-http-request.js:252:48)
    at ClientRequest.emit (node:events:527:28)
    at Socket.socketErrorListener (node:_http_client:454:9)
    at Socket.emit (node:events:527:28)
    at emitErrorNT (node:internal/streams/destroy:157:8)
    at emitErrorCloseNT (node:internal/streams/destroy:122:3)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
```

**Comment by sukanyaparashar** [comment]
Hi @BilliDaGOAT. Thanks for posting this. We are currently working on fixing this issue and we will let you know as soon as it is resolved.

**Comment by haltman-at** [comment]
This should be fixed in the latest Truffle release (5.5.25), so I'm closing this.  Thanks for bringing this up!
